var/spack/repos/builtin/packages/neurodamus/package.py

```diff
@@ -87,11 +89,13 @@ def do_stage(self, mirror_only=False):
     # additional libraries required while linking synapse tool
     def syntool_dep_libs(self):
         spec = self.spec
-        ld_flags = spec['synapsetool'].libs.ld_flags
-        for lib in ['libboost_system-mt', 'libboost_filesystem-mt']:
-            libs = find_libraries(lib, root=spec['boost'].prefix,
-                                  shared=False, recursive=True)
-            ld_flags += ' %s ' % (libs.ld_flags)
+        shared = spec['synapsetool'].satisfies("+shared")
+        ld_flags = find_libraries('libsyn2', spec["synapsetool"].prefix, shared, True).ld_flags
+        if not shared:
+            for lib in ['libboost_system-mt', 'libboost_filesystem-mt']:
+                ld_flags += ' ' + find_libraries(lib, spec['boost'].prefix, False, True).ld_flags
+            if spec['synapsetool'].satisfies("+sonata"):
+                ld_flags += ' ' + find_libraries('libsonata', spec['libsonata'].prefix, False, True).ld_flags
```

Comment:
I tried to do something along these lines, but `ldd`-ing `special` binary exposed that `libsonata` is still dynamically linked (as is Boost, btw).
Is it intended?

Reply:
The ld_flags end up being generic. e.g.
```
-L/gpfs/bbp.cscs.ch/apps/hpc/spack-deployments/18-06-2018/install/linux-rhel7-x86_64/gcc-6.4.0/boost-1.67.0-6skhmz/lib -lboost_system-mt -L/gpfs/bbp.cscs.ch/apps/hpc/spack-deployments/18-06-2018/install/linux-rhel7-x86_64/gcc-6.4.0/boost-1.67.0-6skhmz/lib -lboost_filesystem-mt -L/gpfs/bbp.cscs.ch/home/leite/test/blueconfigs/INSTALL_HOME/install/linux-rhel7-x86_64/intel-18.0.1/libsonata-develop-fhcp6s/lib -lsonata
```
I wonder if it's a spack bug, of it spack leaves it generic so that the linker could fall back to shared, and for some reason that is happening.

Reply:
Since we are lookong for static libs anyway what about we use the full path?